### DISCUSSION TITLE
Using RootEncoder to stream content from video application (eg livetv) to RTSP server

### DISCUSSION THREAD

**Discussion by mph070770** [pedroSG94/RootEncoder#1443] [Q&A] [answered]
Hi.
In my first test, I have root encoder working and can stream from my android tablet to the mediamtx server using RTSP.  I can then view the stream using VLC.  

I have a number of usecases where I want to stream liveTV (such as ITVX here in the UK) to multiple clients around my house (that don't have the option of ITVX but can receive streams).  When I launch the liveTV app, the VLC screen goes blank until I stop the liveTV stream.  Is there a way to make this work, and use the audio from the application rather than the microphone?  I have modest Android programming skills so can hopefully make any suggested edits to the demo code.

Thanks in advance.

**Comment by pedroSG94** [comment, marked as answer]
Hello,

I think it is normal. MediaProjection class is the only way to record your screen content and you can avoid that others apps can use this class to record your content:
https://stackoverflow.com/questions/48800984/how-to-disable-taking-screen-shot-in-android
So this could be intentional by ITVX app to avoid illegal use of the content. Other apps like Netflix or Amazon Prime Video use this too. You can test it opening the app in your device and try to take a screenshoot. The video content should be blank like you describe.

About record audio of the device. You can do it using InternalSource  if you are using StreamBase class (the usage is the same that ScreenSource, you can even use the same MediaProjection than ScreenSource) or prepareInternalAudio if you are using DisplayBase class but this have exactly the same problem that record the screen content because this depend of MediaProjection class too.

**Comment by mph070770** [comment]
Thanks @pedroSG94 , I guess that makes sense.  It's a shame.
